Remotion/Data/DomainObjects.UnitTests/Tracing/TracingDbCommandTest.cs

```diff
@@ -369,7 +496,68 @@ public void ExecuteReaderWithOverload ()
           .Verifiable();
       _innerCommandMock
           .InVerifiableSequence(sequence)
-          .Setup(mock => mock.ExecuteReader(CommandBehavior.SchemaOnly))
+          .Protected()
+          .Setup<DbDataReader>("ExecuteDbDataReader", CommandBehavior.SchemaOnly)
+          .Throws(exception)
+          .Verifiable();
+
+      _extensionMock.InVerifiableSequence(sequence).Setup(mock => mock.QueryError(_connectionID, _command.QueryID, exception)).Verifiable();
+
+      Assert.That(() => _command.ExecuteReader(CommandBehavior.SchemaOnly), Throws.Exception.SameAs(exception));
+
+      _innerCommandMock.Verify();
+      _extensionMock.Verify();
+      sequence.Verify();
+    }
+
+    [Test]
+    public void ExecuteReaderWithPublicImplementation ()
+    {
+      var readerStub = new Mock<DbDataReader>();
+      _innerCommandMock.Setup(mock => mock.CommandText).Returns("commandText");
+      _innerCommandMock.Protected().Setup<DbParameterCollection>("DbParameterCollection").Returns(CreateParameterCollection());
+
+      var sequence = new VerifiableSequence();
+      _extensionMock
+          .InVerifiableSequence(sequence)
+          .Setup(mock => mock.QueryExecuting(_connectionID, _command.QueryID, "commandText", It.IsNotNull<IDictionary<string, object>>()))
+          .Verifiable();
+      _innerCommandMock.InVerifiableSequence(sequence).Protected().Setup<DbDataReader>("ExecuteDbDataReader", CommandBehavior.Default).Returns(readerStub.Object).Verifiable();
+
+      _extensionMock
+          .InVerifiableSequence(sequence)
+          .Setup(mock => mock.QueryExecuted(_connectionID, _command.QueryID, It.Is<TimeSpan>(_ => _ > TimeSpan.Zero)))
+          .Verifiable();
+
+      DbDataReader actualReader = ((DbCommand)_command).ExecuteReader();
+
+      Assert.That(actualReader, Is.InstanceOf(typeof(TracingDataReader)));
+      Assert.That(((TracingDataReader)actualReader).WrappedInstance, Is.SameAs(readerStub.Object));
+      Assert.That(((TracingDataReader)actualReader).ConnectionID, Is.EqualTo(_connectionID));
+      Assert.That(((TracingDataReader)actualReader).QueryID, Is.EqualTo(_command.QueryID));
+      Assert.That(((TracingDataReader)actualReader).PersistenceExtension, Is.SameAs(_extensionMock.Object));
+
+      _innerCommandMock.Verify();
+      _extensionMock.Verify();
+      sequence.Verify();
+    }
+
+    [Test]
+    public void ExecuteReader_WithPublicImplementationAndCommandBehavior ()
```

Comment:
```suggestion
    public void ExecuteReaderWithPublicImplementationAndCommandBehavior ()
```
consitency with rest of file. Since we touch so many lines anyway, you could also inject the `_` into all the `With`-names